walnut/converter/converter.go

```diff
@@ -0,0 +1,156 @@
+/*
+ * === This file is part of ALICE O² ===
+ *
+ * Copyright 2020 CERN and copyright holders of ALICE O².
+ * Author: Ayaan Zaidi <azaidi@cern.ch>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * In applying this license CERN does not waive the privileges and
+ * immunities granted to it by virtue of its status as an
+ * Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+package converter
+
+import (
+	"fmt"
+	"gopkg.in/yaml.v3"
+	"io/ioutil"
+	"os"
+
+	"github.com/AliceO2Group/Control/common"
+	"github.com/AliceO2Group/Control/common/controlmode"
+	"github.com/AliceO2Group/Control/common/gera"
+	"github.com/AliceO2Group/Control/core/task"
+	"github.com/AliceO2Group/Control/core/task/channel"
+)
+
+// return pointer to float64
+func createFloat(x float64) *float64 {
+	return &x
+}
+
+// return pointer to bool
+func createBool(x bool) *bool {
+	return &x
+}
+
+// return pointer to string
+func createString(x string) *string {
+	return &x
+}
+
+// ExtractTaskClasses takes in a DPL Dump string and extracts
+// an array of Tasks
+func ExtractTaskClasses(DPL Dump) (tasks []*task.Class, err error) {
+
+	for index := range DPL.Workflows {
+		var channelName string
+		if index+1 == len(DPL.Workflows) {
```

Comment:
I've noticed you use `index+1` multiple places, looks like a magic value. What's its significance, just navigating through components of the DPL dump? Perhaps it could be a variable instead of doing `index+1` multiple times.

Reply:
The items in the `workflows` array begin with the first `task`. The items in the metadata array additionally have an `internal-dpl-clock` item preceding all the others. This index value is simply an offset to match the correct `workflow` with it's `metadata`. I'll add a variable.

Reply:
`internal-dpl-clock` is a task like any other, it should not receive special treatment. If/when DPL drops it, it will simply not appear in dumps any more.

Reply:
Since `internal-dpl-clock` is only present inside the `metadata` array and not the `workflow` array of the DPL dump, the current implementation does not include it in the import. I'm not sure if this a bug or an intentional design choice.

@ktf since `internal-dpl-clock` is a task, should it not be included in the `workflow` array?

Reply:
@ktf could you please comment on this? It seems like `internal-dpl-clock` is treated differently from other tasks in the DPL dump, and only appears in the `metadata` section. Is this an inconsistency of the DPL dump output, or are we missing some understanding of the DPL dump format?